salt/modules/openvswitch.py

```diff
@@ -0,0 +1,302 @@
+# -*- coding: utf-8 -*-
+"""
+Support for Open vSwitch
+"""
+
+# Import python libs
+import logging
+
+# Import salt libs
+import salt.utils
+
+log = logging.getLogger(__name__)
+
+
+def __virtual__():
+    """
+    Only load the module if Open vSwitch is installed
+    """
+    cmd = _detect_os()
+    if salt.utils.which(cmd):
+        return 'openvswitch'
+    return False
+
+
+def _detect_os():
+    """
+    Open vSwitch command may differ depending on packaging.
+    """
+    return 'ovs-vsctl'
+
+
+def _param_may_exist(may_exist):
+    """
+    Returns --may-exist parameter for Open vSwitch command.
+
+    Args:
+        may_exist: Boolean whether to use this parameter.
+
+    Returns:
+        String '--may-exist ' or empty string.
+    """
+    if may_exist:
+        return '--may-exist '
+    else:
+        return ''
+
+
+def _param_if_exists(if_exists):
+    """
+    Returns --if-exist parameter for Open vSwitch command.
+
+    Args:
+        if_exists: Boolean whether to use this parameter.
+
+    Returns:
+        String '--if-exist ' or empty string.
+    """
+    if if_exists:
+        return '--if-exists '
+    else:
+        return ''
+
+
+def _retcode_to_bool(retcode):
+    """
+    Evaulates Open vSwitch command`s retcode value.
+
+    Args:
+        retcode: Value of retcode field from response, should be 0, 1 or 2.
+
+    Returns:
+        True on 0, else False
+    """
+    if retcode == 0:
+        return True
+    else:
+        return False
+
```

Comment:
The above will return a bool. It's just stylistic though so if you want to keep what you have, that is fine with me.